docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,694 @@
+    extension:
+      type: Lambda 
+      config: # type JSON
+        arn: arn:aws:lambda:us-west-2:123456789012:function:my-function
+    endpoints:
+      - address: 1.1.1.1
+        port: 12345
+      - address: example.com
+      - address: unix://....
+    tls:
+      version:
+        min: TLSv1_2 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+        max: TLSv1_3 # or TLS_AUTO, TLSv1_0, TLSv1_1, TLSv1_2, TLSv1_3
+      allowRenegotiation: false
+      verification:
+        skipSAN: true # if this is true then subjectAltNames don't take effect
+        subjectAltNames: # if subjectAltNames is not defined then take domain or ips
+          - example.com
+          - "spiffe://example.local/ns/local"
+        caCert: 
+          inline: 123
+        clientCert:
+          secret: 123
+        clientKey:
+          secret: 123
+status: 
+  vip:
+    value: 242.0.0.1
+    type: Kuma
+  addresses:
+  - hostname: example.ext.svc.local
+    status: NotAvailable
+    origin:
+      kind: HostnameGenerator
+      name: k8s-example-service-hostname
+    reason: "addresses are overlapping with ext2"
+```
+* **spec**:
+  * **match**: defines a traffic that should be routed through the sidecar
+    * **type**: type of the match, one of `KumaHostname`, `Domain`, `CIDR` and `IP` are available
+      * `KumaHostname`: matches hostnames provided by HostnameGenerator
+      * `Domain`: handles traffic to the specified domain
+      * `CIDR`: handles traffic to specified addresses range
+      * `IP` handles the traffic to specified IP
+    * **value**: depends of the type can be an existing domain, CIDR or IP. Not available when type `KumaHostname`
+    * **port**: defines a port to which a user does requests
+    * **protocol**: defines a protocol of the communication. Possible values:
+      * `tcp`: WARNING: shouldn't be used when match is `Domain` type. On the TCP level we are not able to disinguish domain, in this case it is going to hijack whole traffic on this port. We are going to validate configuration and do not apply config when protocol is tcp and `type: Domain`.
+      * `grpc`
+      * `http`
+      * `http2`
+  * **type**: defines what kind of destination it is, currently `Managed`
+    * `Managed`: allows creating a set of destination endpoints and `TLS` configuration
+  * **managed**: defines where matched requests should be routed, only for a `type: Managed`
+    * **extension**: struct for a plugin configuration
+      * **type**: defines what kind of plugin to use, it's a string type so any new plugins should works.
+      * **config**: json map that is mapped to configuration provided in the type.
+    * **endpoints**: defines a list of endpoints.
+      * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+      * **port**: defines a port of a destination.
+    * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
```

Comment:
There's no longer any indication that TLS should be used. How do you indicate to Envoy that it should initiate the connection with or without tls when using all defaults (which seems equivalent to not setting `tls`).

Reply:
I'm restoring `enabled` - not 100% sure why it was removed

Reply:
let's go for `enabled`